tests/examples_arguments_syntax/interactive_reorder_stacked_bars.py

```diff
@@ -0,0 +1,31 @@
+"""
+Reorder stacked bar segments
+============================
+This example uses a calculate transform
+to check the values of the "site" column
+vs the clicked values in the legend,
+and assigns a lower order (0)
+if there is a match.
+The use of "indexOf" checks for equality in an array,
+which here allows for multiple segments to be reordered
+by holding down the shift key while clicking the legend.
+"""
+# category: interactive charts
+import altair as alt
+from vega_datasets import data
+
+selection = alt.selection_point(fields=['site'], bind='legend')
+
+source = data.barley.url
+
+alt.Chart(source).mark_bar().transform_calculate(
+    site_order=f"if({selection.name}.site && indexof({selection.name}.site, datum.site) !== -1, 0, 1)"
```

Comment:
Im fine with current approach, but an observation is that this is JavaScript syntax. I'm not sure if the Python syntax for this type of Vega expressions is more readable. 

While often underestimated in efficiency, we don't really have a good approach documented to work with these Vega expressions.

Reply:
Yes, @jonmmease made a similar comment in https://github.com/altair-viz/altair/pull/3362#discussion_r1526139226 and I would also prefer to change to the Python syntax in general in the docs. However, I think we should solve https://github.com/altair-viz/altair/issues/3366 first, so that the Py syntax covers everything that the JS syntax can do (this instance is the only difference I'm aware of).

Reply:
Ah, yes, I remember being bitten by this recently too!